### PR TITLE
Youtube update

### PR DESCRIPTION
**Pull request by Bas950**

### CODE REVIEW

File: websites/Y/YouTube/presence.ts [hunk 1]
```diff
@@ -251,7 +359,7 @@ presence.on("UpdateData", async () => {
       document.location.pathname.includes("/user")
     ) {
       //Sometimes causes problems
-
+      let user;
```
Comment (ririxidev) [inline websites/Y/YouTube/presence.ts:362]: ``user: string`` kek

File: websites/Y/YouTube/presence.ts [hunk 2]
```diff
@@ -281,112 +389,134 @@ presence.on("UpdateData", async () => {
...
-      presenceData.state = "their purchases"; //youtube.purchases
+      presenceData.details = (await strings).browsingThrough;
+      presenceData.state = (await strings).purchases;
       presenceData.startTimestamp = browsingStamp;
     } else if (document.location.pathname.includes("/playlist")) {
-      presenceData.details = "Viewing playlist:"; //general.viewPlaylist
+      presenceData.details = (await strings).viewPlaylist;
 
-      title = document.querySelector("#text-displayed");
+      let title = document.querySelector("#text-displayed");
```
Comment (ririxidev) [inline websites/Y/YouTube/presence.ts:452]: ``title: string | null``

File: websites/Y/YouTube/presence.ts [hunk 3]
```diff
@@ -395,52 +525,63 @@ presence.on("UpdateData", async () => {
...
+      },
+      browsingStamp = Math.floor(Date.now() / 1000);
 
     if (document.location.pathname.includes("/videos")) {
-      presenceData.details = "Viewing their videos"; //youtube.studio.viewVideos
+      presenceData.details = (await strings).studioVid;
       presenceData.startTimestamp = browsingStamp;
     } else if (document.location.pathname.includes("/video")) {
-      title = document.querySelector("#entity-name");
+      const title = document.querySelector("#entity-name");
```
Comment (ririxidev) [inline websites/Y/YouTube/presence.ts:538]: ``title: string``

File: websites/Y/YouTube/presence.ts [hunk 4]
```diff
@@ -227,22 +335,22 @@ presence.on("UpdateData", async () => {
...
-      largeImageKey: "yt_lg"
-    };
-
-    var search: any;
-    var user: any;
-    var browsingStamp = Math.floor(Date.now() / 1000);
+        largeImageKey: "yt_lg"
+      },
+      browsingStamp = Math.floor(Date.now() / 1000);
+    let searching = false;
```
Comment (ririxidev) [inline websites/Y/YouTube/presence.ts:341]: ``searching: boolean``

**Review by ririxidev — APPROVED** [review]
kek

**Review by ririxidev — APPROVED** [review]

File: websites/Y/YouTube/dist/metadata.json [hunk 5]
```diff
@@ -29,13 +29,51 @@
     "www.youtube.com",
     "studio.youtube.com"
   ],
-  "version": "4.2.7",
+  "version": "5.0.0",
```
Comment (eggsy) [inline websites/Y/YouTube/dist/metadata.json:32]: Why jump to 5?!?! Just use 4.3.0

**Review by eggsy — CHANGES_REQUESTED** [review]

File: websites/Y/YouTube/dist/metadata.json (hunk 5 above)
Comment (ririxidev) [reply websites/Y/YouTube/dist/metadata.json:32]: bruh he rewrote whole presence and you are saying jump to 4.3.0 xD

Comment (eggsy) [reply websites/Y/YouTube/dist/metadata.json:32]: And me rewrote my entire website

Comment (Bas950) [reply websites/Y/YouTube/dist/metadata.json:32]: This repo is about PreMiD Presences not your website.

Comment (eggsy) [reply websites/Y/YouTube/dist/metadata.json:32]: Still, it's a big change for the service the presence is made for.

**Pull request merged**